pthread/pthread.c

```diff
@@ -482,45 +605,136 @@ int pthread_setschedparam(pthread_t thread, int policy,
 
 int pthread_mutex_init(pthread_mutex_t *mutex, const pthread_mutexattr_t *attr)
 {
+	int err;
+
 	(void)attr;
 
-	return mutexCreate(mutex);
+	err = mutexCreate(&mutex->mutexh);
+	if (err == EOK) {
+		mutex->initialized = 1;
+	}
+	return err;
 }
 
 
 int pthread_mutex_lock(pthread_mutex_t *mutex)
 {
-	if (mutex == NULL)
-		return -EINVAL;
+	int err = EOK;
+
+	if (mutex->initialized == 0) {
+		mutexLock(pthread_common.mutex_cond_init_lock);
+		if (mutex->initialized == 0) {
+			err = mutexCreate(&mutex->mutexh);
+			if (err == EOK) {
+				mutex->initialized = 1;
+			}
+		}
+		mutexUnlock(pthread_common.mutex_cond_init_lock);
+	}
+	if (err == EOK) {
+		err = mutexLock(mutex->mutexh);
+	}
 
-	return mutexLock(*mutex);
+	return err;
 }
 
 
 int pthread_mutex_trylock(pthread_mutex_t *mutex)
 {
-	if (mutex == NULL)
-		return -EINVAL;
+	int err = EOK;
 
-	return mutexTry(*mutex);
+	if (mutex->initialized == 0) {
+		mutexLock(pthread_common.mutex_cond_init_lock);
+		if (mutex->initialized == 0) {
+			err = mutexCreate(&mutex->mutexh);
+			if (err == EOK) {
+				mutex->initialized = 1;
+			}
+		}
+		mutexUnlock(pthread_common.mutex_cond_init_lock);
+	}
+	if (err == EOK) {
+		err = mutexTry(mutex->mutexh);
+	}
+
+	return err;
 }
 
 
 int pthread_mutex_unlock(pthread_mutex_t *mutex)
 {
-	if (mutex == NULL)
-		return -EINVAL;
+	int err = EOK;
+
+	if (mutex->initialized == 0) {
+		mutexLock(pthread_common.mutex_cond_init_lock);
+		if (mutex->initialized == 0) {
+			err = mutexCreate(&mutex->mutexh);
+			if (err == EOK) {
+				mutex->initialized = 1;
+			}
+		}
+		mutexUnlock(pthread_common.mutex_cond_init_lock);
+	}
+	if (err == EOK) {
+		err = mutexUnlock(mutex->mutexh);
+	}
 
-	return mutexUnlock(*mutex);
+	return err;
 }
 
 
 int pthread_mutex_destroy(pthread_mutex_t *mutex)
 {
-	if (mutex == NULL)
-		return -EINVAL;
+	int err = EOK;
+
+	mutexLock(pthread_common.mutex_cond_init_lock);
+	if (mutex->initialized == 1) {
+		mutexUnlock(pthread_common.mutex_cond_init_lock);
+		err = resourceDestroy(mutex->mutexh);
+	}
+
+	return err;
+}
+
+
+int pthread_mutexattr_destroy(pthread_mutexattr_t *attr)
+{
+	return EOK;
+}
+
+
+int pthread_mutexattr_init(pthread_mutexattr_t *attr)
+{
+	attr->type = PTHREAD_MUTEX_DEFAULT;
+	return EOK;
+}
+
+
+int pthread_mutexattr_gettype(const pthread_mutexattr_t *attr, int *type)
+{
+	*type = attr->type;
+	return EOK;
+}
+
+
+int pthread_mutexattr_settype(pthread_mutexattr_t *attr, int type)
+{
+	int err = EOK;
 
-	return resourceDestroy(*mutex);
+	if (type != PTHREAD_MUTEX_DEFAULT && type != PTHREAD_MUTEX_ERRORCHECK &&
```

Comment:
You could invert this condition (using de morgan's law) and only on valid params do `attr->type = type` and `err = EOK`, this way there's no need for `else`